### ISSUE TITLE
Crash with Clear Area tool

### ISSUE THREAD

**Issue by LeeSpork** [OpenLoco/OpenLoco#2654]
**Version information:**
 - Platform: Windows 10
 - Version: 24.09.1
 - Build/commit: 4fe5ae37

I wanted to destroy this road that was placed by an AI that had gone bankrupt so that I could build a railway, and I mistakenly tried using the Clear Area tool instead of the road tool. I tried both left clicking and right clicking, and the land underneath the road was cleared of snow, and then the game crashed.

I was able to replicate the crash by loading the attached autosave and doing that again, hence the two dump files and two screenshots.

**Dump file and most recent save file**
[crash dmps and most recent autosave.zip](https://github.com/user-attachments/files/17339868/crash.dmps.and.most.recent.autosave.zip)

**Screenshots**
![Image](https://github.com/user-attachments/assets/1f9cb587-5641-45fb-a126-c3ba60256c6c)
![Image](https://github.com/user-attachments/assets/4e0ff904-760c-4446-bb19-9e59d177e12d)

**Comment by LeeSpork** [comment]
I renamed and edited the issue because I realized I was confused and mistaken in what was causing it, I hope that is okay.

**Comment by LeeSpork** [comment]
Also crashes in original Locomotion:
![Image](https://github.com/user-attachments/assets/d9103720-7680-446b-8043-13b6a3006618)

**Comment by duncanspumpkin** [comment]
In vanilla does the ai sell all its road when it goes bankrupt?

The issue is that the road is still there and it has an owner which no longer exists. When trying to draw the tooltip about the road it tries to get the owner and fails. I'm not sure how vanilla did this though. I assume in vanilla the ai would have sold the road?

**Comment by LeeSpork** [comment]
Yes. I created a save file to test this, by using OpenLoco's cheats menu to set a company to bankrupt: [save file.zip](https://github.com/user-attachments/files/17607945/save.file.zip)

Fast forward about 9 months and: In original Locomotion, the AI's road will be removed about a month before the company gets deleted, while in OpenLoco, it will remain (making this crash possible).

[Screenshots removed as they were not relevant]

**Comment by duncanspumpkin** [comment]
That wouldn't be a safe enough test I want to check if a fully vanilla game can get in this situation by starting the save in OpenLoco its invalidating the test.

**Comment by LeeSpork** [comment]
Here is a save I created fully with original Locomotion (Steam): [One Farm with Competing Trucks 0017-1.zip](https://github.com/user-attachments/files/20513651/One.Farm.with.Competing.Trucks.0017-1.zip)

Right after loading this save file, all roads owned by Scarlet Freight get removed in original Locomotion, and in OpenLoco v24.01.1 and bellow.

In OpenLoco 24.02 and above, I think I can faintly hear construction sounds?, but the roads do not get removed. A few days later, the company is deleted, and hovering over the roads causes the crash.
